ns1/datadog_checks/ns1/check.py

```diff
@@ -0,0 +1,448 @@
+                if res and len(res) > 0:
+                    percent_available = res[0][1]
+                    return percent_available, True
+                else:
+                    return None, False
+        except Exception:
+            return None, False
+
+    def extract_peak_lps(self, jsonResult):
+        try:
+            graph = jsonResult[0]["graph"]
+            res = sorted(graph, key=lambda x: x[0], reverse=True)
+            curr_lps = res[0][1]
+            logmsg = 'extract_peak_lps:{lps}'.format(lps=curr_lps)
+            logging.info(logmsg)
+            return curr_lps, True
+
+        except Exception as e:
+            logging.info(e)
+            return None, False
+
+    def extract_usage_count(self, key, jsonResult):
+
+        try:
+            graph = jsonResult[0]["graph"]
+            # usage api will return array of dictionaries, we want to get 'graph' object
+            # which in turn is list of lists, each element being [timestamp, query_count]
+            # so, get last query count from result.
+            # Sort by timestamp descending order to make sure we get latest
+            res = sorted(graph, key=lambda x: x[0], reverse=True)
+
+            curr_timestamp = res[0][0]
+            curr_count = res[0][1]
+            # find this metric in usage count
+            if key in self.usage_count:
+                prev_timestamp = self.usage_count[key][0]
+                prev_count = self.usage_count[key][1]
+                if curr_timestamp == prev_timestamp:
+                    if curr_count >= prev_count:
+                        self.usage_count[key] = [prev_timestamp, curr_count]
+                        result = curr_count - prev_count
+                    else:
+                        result = 0
+                else:
+                    self.usage_count[key] = [curr_timestamp, curr_count]
+                    result = curr_count
+            else:
+                self.usage_count[key] = [curr_timestamp, curr_count]
+                result = curr_count
+
+            return result, True
+        except Exception:
+            return None, False
+
+    def extract_records_ttl(self, jsonResult):
+        try:
+            zoneTtl = {}
+            for zone in jsonResult["records"]:
+                zoneTtl[zone["domain"]] = zone["ttl"]
+            return zoneTtl, True
+        except Exception:
+            return None, False
+
+    def extract_billing(self, jsonResult):
+        try:
+            billing = {}
+            billing["usage"] = jsonResult["totals"]["queries"]
+            billing["limit"] = jsonResult["any"]["query_credit"]
+            return billing, True
+        except Exception:
+            return None, False
+
+    def get_stats(self, url):
+        # Perform HTTP Requests with our HTTP wrapper.
+        # More info at https://datadoghq.dev/integrations-core/base/http/
+        try:
+            response = self.http.get(url, headers=self.headers)
```

Comment:
```suggestion
            response = self.http.get(url, extra_headers=self.headers)
```
If you don't want to remove the default agent headers.